src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -261,8 +261,63 @@ namespace CDASimAdapter{
         JSONCPP_STRING err;
         if(!reader->parse(json_str.c_str(), json_str.c_str() + json_str.length(), &json_v, &err))
         {
-            PLOG(logERROR) << "Error parsing sensors from string: " << json_str << std::endl;
+            throw std::invalid_argument("Error parsing JSON from string: " + json_str);
         }
         return json_v;
     }
+
+    Json::Value CDASimConnection::read_sensor_configuration_file(const std::string &file_path) const {
+        // Sensor Configuration File JSON format
+        /*  {
+            "sensorId": "sensor_1",
+            "type": "SemanticLidar",
+            "ref": {
+                "type": "CARTESIAN",
+                "location": {
+                    "x": 1.0, 
+                    "y": 2.0, 
+                    "z": -3.2 
+                },
+                "orientation": { 
+                    "yaw": 0.0,
+                    "pitch": 0.0,
+                    "roll": 0.0
+                    }
+            }
+        } 
+        ] */
+        auto sensor_configuration = read_json_file(file_path);
+        Json::Value sensors_registration;
+        for ( int index = 0; index < sensor_configuration.size(); ++index ) {
+            Json::Value sensor;
+            sensor["sensorId"] = sensor_configuration[index]["sensorId"];
```

Comment:
What happens if sensorId is missing in the JSON file?

Reply:
We get a warning that the file can not be found and we send a infrastructure registration without sensor information